### ISSUE TITLE
Fail to get the checkbox of the winform DateTimePicker

### ISSUE THREAD

**Issue by stg609** [TestStack/White#500]
![image](https://cloud.githubusercontent.com/assets/6631727/21167281/73c6e144-c1e6-11e6-92eb-00cb8cb304f1.png)

I tried many ways to get the checkbox of the DateTimePicker control, so that I can set the state to checked or unchecked programmatically. But all failed.
Try 1, dtp.AutomationElement.GetCurrentPattern(TogglePattern.Pattern) , but it return null.
Try 2, dtp.AutomationElement.FindAll(), can't find the checkbox too.

BTW, I can't find any information about that checkbox from Inspect.

Is there anyway to get the checkbox?

**Comment by maninwindow** [comment]
@stg609  Have you tried `currentWindow.GetMultiple(SearchCriteria.ByControlType(ControlType.CheckBox));` ?

**Comment by Roemer** [comment]
Unfortunately, the DateTimePicker seems to not expose the checkbox in any way. What you probably could do is create your own DateTimePicker control (which inherits the DateTimePicker) and override the CreateAccessibilityInstance to return your own AccessibleObject which somehow exposes a checkbox with the correct checked state or which exposes the additional TogglePattern.

**Comment by Roemer** [comment]
Here's some code to give you a quick start. It basically just adds a checkbox which reflects the value. Unfortunately this checkbox needs to be visible and have a size or UIA will somehow not display it. There are possible better methods but at least it works.
```csharp
public class MyDate : DateTimePicker
{
    internal readonly CheckBox CheckBox;

    public MyDate()
    {
        CheckBox = new CheckBox();
        CheckBox.Parent = this;
        CheckBox.Visible = true;
        CheckBox.Width = 1;
        CheckBox.Height = 1;
        CheckBox.Checked = Checked;
    }

    protected override void OnValueChanged(EventArgs eventargs)
    {
        base.OnValueChanged(eventargs);
        CheckBox.Checked = Checked;
    }

    protected override AccessibleObject CreateAccessibilityInstance()
    {
        return new MyDateAccessibleObject(this);
    }
}

public sealed class MyDateAccessibleObject : DateTimePicker.DateTimePickerAccessibleObject
{
    private readonly MyDate _ownerControl;

    public MyDateAccessibleObject(MyDate ownerControl) : base(ownerControl)
    {
        _ownerControl = ownerControl;
    }

    public override int GetChildCount()
    {
        return 1;
    }

    public override AccessibleObject GetChild(int index)
    {
        return _ownerControl.CheckBox.AccessibilityObject;
    }
}
```

**Comment by stg609** [comment]
@maninwindow Yes, I have tried that way, and got nothing.

**Comment by stg609** [comment]
@Roemer Thanks for the suggestion and the code, it's very helpful. However, I can't simply replace the DateTimePicker with myDate as I don't have the control over that application.

Actually, I've found there's a legacyIAccessible.State property of the DateTimePicker. But I don't know how to set the value as I want. 

![image](https://cloud.githubusercontent.com/assets/6631727/21178127/176c2fec-c229-11e6-9be9-20ccd3f2dea1.png)

To make use of the legacy pattern, I added a reference to the UIAComWrapper.dll and used below code to get the state:
`var pattern = ((LegacyIAccessiblePattern) child.GetCurrentPattern(LegacyIAccessiblePattern.Pattern));
var state = pattern.GetIAccessible().accState;`

However, the GetIAccessible() return null. And I can't find much useful information talking about it.

The last option I can think about is to send a window's message directly to the control.
